### PR TITLE
Check DataTypes solution

### PR DESCRIPTION
**Pull request by TMatrix**

### CODE REVIEW

File: Exercises/3-by-reference.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
-const inc = null;
+const inc = function (obj) {
```
Comment (tshemsedinov) [inline Exercises/3-by-reference.js]:
```suggestion
const inc = obj => {
```

File: Exercises/4-count-types.js [hunk 2]
```diff
@@ -1,5 +1,15 @@
 'use strict';
 
-const countTypesInArray = null;
+const countTypesInArray = arr => {
+  const types = {};
+  for (const el of arr) {
+    const t = el === null ? 'null' : typeof el;
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]:
```suggestion
    const type = el === null ? 'null' : typeof el;
```

File: Exercises/4-count-types.js [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    if (!(t in types)) {
+      types[t] = 0;
+    }
+    types[t] += 1;
+  }
+  return types;
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]: See implementations here https://github.com/HowProgrammingWorks/DataTypes/pull/21#discussion_r324387734

Comment (TMatrix) [reply Exercises/4-count-types.js]: Thank you for reference to more elegance code

**Comment by tshemsedinov** [comment]
:+1:

**Pull request closed without merging**